tutorials/3d/3d_antialiasing.rst

```diff
@@ -0,0 +1,237 @@
+.. _doc_3d_antialiasing:
+
+3D antialiasing
+===============
+
+.. Images on this page were generated using the project below
+.. (except for `antialiasing_none_scaled.png`):
+.. https://github.com/Calinou/godot-antialiasing-comparison
+
+.. seealso::
+
+    Godot also supports antialiasing in 2D rendering. This is covered on the
+    :ref:`doc_2d_antialiasing` page.
+
+Introduction
+------------
+
+Due to their limited resolution, scenes rendered in 3D can exhibit aliasing
+artifacts. These artifacts manifest in the form of a "staircase" effect on
+geometry edges, but also with reflective surfaces flickering in and out
+(specular aliasing).
+
+In the example below, you can notice how
+edges have a blocky appearance. The vegetation is also flickering in and out,
+and thin lines on top of the box have almost disappeared:
+
+.. figure:: img/antialiasing_none_scaled.png
+   :alt: Image is scaled by 2× with nearest-neighbor filtering to make aliasing more noticeable.
+   :align: center
+
+   Image is scaled by 2× with nearest-neighbor filtering to make aliasing more noticeable.
+
+To combat this, various antialiasing techniques can be used in Godot. These are
+detailed below.
+
+Multisample antialiasing (MSAA)
+-------------------------------
+
+This technique is the "historical" way of dealing with aliasing. MSAA is very
+effective on geometry edges (especially at higher levels). MSAA does not
+introduce any blurriness whatsoever.
+
+MSAA in available in 3 levels: 2×, 4×, 8×. Higher levels are more effective at
+antialiasing edges, but are significantly more demanding. In games with modern
+visuals, sticking to 2× or 4× MSAA is highly recommended as 8× MSAA is usually
+too demanding.
+
+The downside of MSAA is that it only operates on edges. This is because MSAA
+increases the number of *coverage* samples, but not the number of *color*
+samples. Using multiple coverage samples allows the coverage value to be an
+intermediate value between 0 and 1, rather than being limited to 0.0 *or* 1.0.
+The coverage value is used to determine the contribution of the pixel to the
+scene, with 0.0 meaning "no contribution at all" and 1.0 meaning "fully opaque
+pixel".
+
+However, since the number of color samples did not increase, fragment shaders
+are still run for each pixel only once. Therefore, MSAA does not reduce
+transparency aliasing for materials using the **Alpha Scissor** transparency
+mode (1-bit transparency). MSAA is also ineffective on specular aliasing.
```

Comment:
```suggestion
MSAA has the downside of being ineffective on specular aliasing. It also only operates
on edges - it does not reduce transparency aliasing for materials using the **Alpha Scissor**
transparency mode (1-bit transparency).
```

While I like a good delve into technical details, would it be better to keep the section focused only on outlining the downsides of MSAA?

I also tried to move specular aliasing first so that the alpha scissoring problem flows into the next paragraph on mitigating it, but I'm not very happy with my wording.

Reply:
I can remove the description of coverage samples if needed – I'm not sure if it's actually required to understand what's going on.

Reply:
Copying the part I removed for future reference:

```rst
Using multiple coverage samples allows the coverage value to be an
intermediate value between 0 and 1, rather than being limited to 0.0 *or* 1.0.
The coverage value is used to determine the contribution of the pixel to the
scene, with 0.0 meaning "no contribution at all" and 1.0 meaning "fully opaque
pixel".
```